src/ClientOptionsTrait.php

```diff
@@ -115,6 +119,20 @@ private function buildClientOptions(array $options)
         // Keep track of the API Endpoint
         $apiEndpoint = $options['apiEndpoint'] ?? null;
 
+        // If the user provides a logger or the GOOGLE_SDK_DEBUG_LOGGING environment variable is set
+        // we attach a logger to the options array
+        // If we have the logger option SET to null, it should trump the environment variable AKA explicitely
```

Comment:
this looks good, but in the PHPDoc for `ClientOptions` (and here) we should reword this to be slightly more clear, something like
```
setting the "logging" option to `null` will explicitly turn logging off (e.g. a default logger will not be fetched from the environment)
```